Src/Support/Google.Apis.Auth/OAuth2/LocalServerCodeReceiver.cs

```diff
@@ -65,6 +67,69 @@ public class LocalServerCodeReceiver : ICodeReceiver
   </body>
 </html>";
 
+        private static object s_lock = new object();
+        // Current best-guess as to most suitable callback URI.
+        private static string s_callbackUriTemplate;
+        // Has a successful callback been received?
+        private static bool s_receivedCallback;
+
+        /// <summary>
+        /// Create an instance of <see cref="LocalServerCodeReceiver"/>.
+        /// </summary>
+        public LocalServerCodeReceiver()
+        {
+            lock (s_lock)
+            {
+                // Listening on 127.0.0.1 is recommended, but can't be done in non-admin Windows 7 & 8.
+                // So use some tests/heuristics so maybe listen on localhost instead.
+                if (s_callbackUriTemplate != null && !s_receivedCallback)
+                {
+                    // On non-first runs, if a successful callback has not been received
+                    // then force callback to use localhost rather than 127.0.0.1
+                    // This is to heuristically avoid errors on browsers that can't connect to 127.0.0.1
+                    // E.g. IE11 in enhanced protection mode.
+                    s_callbackUriTemplate = CallbackUriTemplateLocalhost;
+                }
+                if (s_callbackUriTemplate == null)
+                {
+                    s_callbackUriTemplate = CallbackUriTemplate127001;
+#if NETSTANDARD1_3
+                    // No check required on NETStandard, it uses TcpListener which can only use IP adddresses, not DNS names.
+#elif NET45
+                    // On first run, check whether it's possible to start a listener on 127.0.0.1
```

Comment:
Maybe add more of a comment to say this listener is not used for anything, hence the immediate disposal?